core/src/main/java/com/google/zxing/aztec/decoder/Decoder.java

```diff
@@ -332,6 +333,26 @@ private static int readCode(boolean[] rawbits, int startIndex, int length) {
     return res;
   }
 
+  private static byte boolsToByte(final boolean[] array, final int start) {
+    byte b = 0;
+    for (int i = 0; i < 8; i++) {
+      if (array[start + i]) {
+        b |= 1 << (7 - i);
+      } else {
+        throw new IndexOutOfBoundsException();
```

Comment:
Ok. Just mean I am not sure why you added this change. It is wrong and makes the test fail. I suggested checking the length of the array. Containing false is not the same thing nor an error. It also needs a simple test to demonstrate and verify the expected behavior